datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Domain/SidebarDomainSection.tsx

```diff
@@ -73,13 +73,14 @@ export const SidebarDomainSection = () => {
                     </>
                 )}
             </div>
-            <SetDomainModal
-                visible={showModal}
-                refetch={refetch}
-                onClose={() => {
-                    setShowModal(false);
-                }}
-            />
+            {showModal && (
+                <SetDomainModal
+                    refetch={refetch}
```

Comment:
We worked on these together - there were unexpected side-effects from not unmounting the modal component and just passing in the `visible` flag. That's why he was needing to do those weird timeouts to blur the input and why values were needing to be reset manually on closing the modal - otherwise all that state would remain in memory and be there when you open the modal again.

The easiest solution where we don't need timeout hacks is to just unmount the whole modal component when it's not visible, so every time it's opened, it's opened fresh without any unexpected side-effects from stale state.